### PR TITLE
Home video page func

### PR DESCRIPTION
**Pull request by irajbhinde**
Hey! Please review my code & let me know if any changes are to be made

Functionalities Added - 

1.  Browsing through Categories on Landing Page
2. Modals on Kebab Icon
3. Modals on Add to playlist
4. Browse through Categories via Chips

Files to check - 

- [ ] src/Components/Cards/Cards.jsx
- [ ] src/Components/Cards/Cards.jsx
- [ ] src/Components/Chips/Chips.jsx
- [ ] src/Components/Playlist Modal/Modal.jsx
- [ ] src/Components/Playlist Modal/modal.css
- [ ] src/Components/context/video-context.js
- [ ] src/Components/reducers/videoReducer.jsx
- [ ] src/Utils/getFilteredData.js
- [ ] src/pages/Video Listing/VideoListingPage.jsx
- [ ] src/pages/Video Listing/video-listing-page.css

GIFY - 

pr_2-vid lib

### CODE REVIEW

File: src/Components/Cards/Cards.jsx [hunk 1]
```diff
@@ -2,62 +2,125 @@ import "./cards.css";
...
-            </div>
-          )
+    <div key={video._id} className="videolisting-cards">
+      <img
+        className="videolisting_img"
+        src="https://i.ytimg.com/vi/f013dR_y7DI/hqdefault.jpg?s…RUAAIhCGAE=&rs=AOn4CLCmmUMogcnMu2KFfSuEnC-AN0plmw"
+        alt="error"
+      />
+      <span className="videolisting-content">
+        <p className="flex_r flex1">{title}</p>
```
Comment (viru2001) [inline src/Components/Cards/Cards.jsx:43]: you can give more generic classname instead of flex_r and flex1

File: src/Components/Cards/Cards.jsx [hunk 2]
```diff
@@ -2,62 +2,125 @@ import "./cards.css";
...
         ({ categoryName, image, bottom }) =>
           !bottom && (
-            <div class="card">
-              <Link to="/videolisting">
-              <img class="category-img" src={image} alt="error" />
-              <p>{categoryName}</p>
-              </Link>
-             
-            </div>
+            <Link to="/videolisting">
```
Comment (nikhil-sharma-nks) [inline src/Components/Cards/Cards.jsx:15]: when rendering components through map you should have key attribute

File: src/Components/Cards/cards.css [hunk 3]
```diff
@@ -92,15 +92,42 @@
...
+.modal {
+  font-size: 1.6rem;
+  position: absolute;
+  right: 0rem;
+  top: 105%;
+  z-index: 2;
+  color: white;
+  width: 26rem;
+  height: max-content;
+  background-color: #202020;
```
Comment (nikhil-sharma-nks) [inline src/Components/Cards/cards.css:118]: if this background color is being used multiple times then you should make it a variable

File: src/Components/Cards/cards.css [hunk 4]
```diff
@@ -92,15 +92,42 @@
...
+}
+
+.modal div {
+  background-color: #202020;
```
Comment (viru2001) [inline src/Components/Cards/cards.css:122]: you can use variable for color here

**Review by nikhil-sharma-nks — APPROVED** [review]
Looks good to merge, just look into the few suggestions. Good work with writing clean and understandable code.

File: src/Components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -1,29 +1,32 @@
...
+import { Link } from "react-router-dom";
 
-export default function Sidebar(){
-    return(
-        <aside class="side-bar">
+export default function Sidebar() {
+  return (
+    <aside class="side-bar">
+      <Link to="/">
         <div>
```
Comment (viru2001) [inline src/Components/Sidebar/Sidebar.jsx:9]: why `<div>` here ? you already used `<Link>`

File: src/Components/reducers/videoReducer.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        },
+      };
+    case "LEARNING":
+      return {
+        ...state,
+        categories: {
+          ...state.categories,
+          Learning: !state.categories.Learning,
+        },
+      };
```
Comment (viru2001) [inline src/Components/reducers/videoReducer.jsx:27]: you should not make different case for every filter, you can optimize this by using only one case for all filters

**Review by viru2001 — APPROVED** [review]
- good folder structure 
- nice ui
gave few suggestions on which you could work on

**Pull request merged**